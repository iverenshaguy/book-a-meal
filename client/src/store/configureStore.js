import { createStore, applyMiddleware, compose } from 'redux';
import { routerMiddleware } from 'connected-react-router';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import rootReducer, { history } from '../reducers/rootReducer';

const enhancers = [];
const middlewares = [thunk, routerMiddleware(history)];


if (process.env.NODE_ENV === 'development') {
  middlewares.push(logger);

  const { devToolsExtension } = window;

  if (typeof devToolsExtension === 'function') {
    enhancers.push(devToolsExtension());
  }
}

const composedEnhancers = compose(applyMiddleware(...middlewares), ...enhancers);

export default createStore(rootReducer, composedEnhancers);
