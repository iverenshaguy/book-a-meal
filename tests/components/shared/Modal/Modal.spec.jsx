import React from 'react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { initialState } from '../../../setup/mockData';
import ModalComponent from '../../../../src/components/shared/Modal';
import ModalContainer from '../../../../src/containers/shared/Modal';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const store = mockStore(initialState);

describe('Modal', () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  it('should render Modal component correctly when type prop is addMeal', () => {
    const shallowWrapper = shallow(<ModalComponent type="addMeal" open submitting={false} toggleModal={jest.fn()} />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('#modal-title-h3').text()).toEqual('Add a Meal');
    expect(shallowWrapper.find('MealModal')).toBeTruthy();
  });

  it('should not render correctly when modal is closed i.e. open prop is false', () => {
    const shallowWrapper = shallow(<ModalComponent type="addMeal" open={false} submitting={false} toggleModal={jest.fn()} />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.text()).toEqual('');
  });

  it('should render correctly when type prop is editMeal', () => {
    const shallowWrapper = shallow(<ModalComponent type="editMeal" open submitting={false} toggleModal={jest.fn()} />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('#modal-title-h3').text()).toEqual('Edit Meal');
    expect(shallowWrapper.find('MealModal')).toBeTruthy();
  });

  it('should render correctly when type prop is deleteMeal', () => {
    const shallowWrapper = shallow(<ModalComponent type="deleteMeal" open submitting={false} toggleModal={jest.fn()} />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('#modal-title-h3').text()).toEqual('Delete Meal');
    expect(shallowWrapper.find('DeleteMealModal')).toBeTruthy();
  });

  it('should render correctly when type prop is deleteSuccessMsg', () => {
    const shallowWrapper = shallow(<ModalComponent type="deleteSuccessMsg" open toggleModal={jest.fn()} />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('#modal-title-h3').text()).toEqual('Delete Meal');
    expect(shallowWrapper.find('p.text-center').text()).toEqual('Meal Deleted Successfully');
  });

  it('should render correctly when type prop is orderSuccessMsg', () => {
    const shallowWrapper = shallow(<ModalComponent type="orderSuccessMsg" open toggleModal={jest.fn()} />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('p.text-center').text()).toEqual('Thank you for your order. Your belly will be filled up shortly');
  });

  it('should render correctly when type prop is orderCanceledMsg', () => {
    const shallowWrapper = shallow(<ModalComponent type="orderCanceledMsg" open toggleModal={jest.fn()} />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('p.text-center').text()).toEqual('Order Canceled Successfully');
  });

  it('should render correctly when type prop is menu', () => {
    const shallowWrapper = shallow(<ModalComponent type="menu" open submitting={false} toggleModal={jest.fn()} />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('#modal-title-h3').text()).toEqual('Select Meal Options');
    expect(shallowWrapper.find('MenuModal')).toBeTruthy();
  });

  it('should render null when type prop is unknown', () => {
    const shallowWrapper = shallow(<ModalComponent type="unknown" open toggleModal={jest.fn()} />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('should call toggleModal function prop when close modal icon is clicked', () => {
    const toggleMock = jest.fn();
    const comp = (<ModalComponent store={store} type="unknown" open submitting={false} toggleModal={toggleMock} />);
    const wrapper = mount(comp);

    wrapper.find('button.link-btn').simulate('click');

    expect(toggleMock).toHaveBeenCalled();
    wrapper.unmount();
  });

  it('should render connected Modal component correctly', () => {
    const comp = (<Provider store={store}><ModalContainer type="addMeal" open toggleModal={jest.fn()} /></Provider>);
    const mountedWrapper = mount(comp);

    expect(toJson(mountedWrapper)).toMatchSnapshot();
    mountedWrapper.unmount();
  });
});
