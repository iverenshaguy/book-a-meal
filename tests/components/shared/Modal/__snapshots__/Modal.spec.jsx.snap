// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should not render correctly when modal is closed i.e. open prop is false 1`] = `""`;

exports[`Modal should render Modal component correctly when type prop is addMeal 1`] = `
<div
  className="modal show"
  id="meal-modal"
>
  <div
    className="modal-content"
  >
    <div
      className="modal-header"
    >
      <div
        className="modal-title"
      >
        <h3
          id="modal-title-h3"
        >
          Add a Meal
        </h3>
      </div>
      <CloseIcon
        btnID="modal-close-icon"
        clickHandler={[Function]}
        divClass={null}
      />
    </div>
    <hr />
    <div
      className="modal-body"
    >
      <MealModal
        meal={null}
        open={true}
        submitting={false}
        toggleModal={[MockFunction]}
        type="addMeal"
      />
    </div>
  </div>
</div>
`;

exports[`Modal should render connected Modal component correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Modal)
    open={true}
    toggleModal={[MockFunction]}
    type="addMeal"
  >
    <Modal
      open={false}
      toggleModal={[Function]}
      type={null}
    />
  </Connect(Modal)>
</Provider>
`;

exports[`Modal should render correctly when type prop is deleteMeal 1`] = `
<div
  className="modal show"
  id="meal-modal"
>
  <div
    className="modal-content"
  >
    <div
      className="modal-header"
    >
      <div
        className="modal-title"
      >
        <h3
          id="modal-title-h3"
        >
          Delete Meal
        </h3>
      </div>
      <CloseIcon
        btnID="modal-close-icon"
        clickHandler={[Function]}
        divClass={null}
      />
    </div>
    <hr />
    <div
      className="modal-body"
    >
      <Connect(DeleteMealModal)
        open={true}
        submitting={false}
        toggleModal={[MockFunction]}
        type="deleteMeal"
      />
    </div>
  </div>
</div>
`;

exports[`Modal should render correctly when type prop is deleteSuccessMsg 1`] = `
<div
  className="modal show"
  id="meal-modal"
>
  <div
    className="modal-content"
  >
    <div
      className="modal-header"
    >
      <div
        className="modal-title"
      >
        <h3
          id="modal-title-h3"
        >
          Delete Meal
        </h3>
      </div>
      <CloseIcon
        btnID="modal-close-icon"
        clickHandler={[Function]}
        divClass={null}
      />
    </div>
    <hr />
    <div
      className="modal-body"
    >
      <p
        className="text-center"
      >
        Meal Deleted Successfully
      </p>
    </div>
  </div>
</div>
`;

exports[`Modal should render correctly when type prop is editMeal 1`] = `
<div
  className="modal show"
  id="meal-modal"
>
  <div
    className="modal-content"
  >
    <div
      className="modal-header"
    >
      <div
        className="modal-title"
      >
        <h3
          id="modal-title-h3"
        >
          Edit Meal
        </h3>
      </div>
      <CloseIcon
        btnID="modal-close-icon"
        clickHandler={[Function]}
        divClass={null}
      />
    </div>
    <hr />
    <div
      className="modal-body"
    >
      <MealModal
        meal={null}
        open={true}
        submitting={false}
        toggleModal={[MockFunction]}
        type="editMeal"
      />
    </div>
  </div>
</div>
`;

exports[`Modal should render correctly when type prop is menu 1`] = `
<div
  className="modal show"
  id="meal-modal"
>
  <div
    className="modal-content"
  >
    <div
      className="modal-header"
    >
      <div
        className="modal-title"
      >
        <h3
          id="modal-title-h3"
        >
          Select Meal Options
        </h3>
      </div>
      <CloseIcon
        btnID="modal-close-icon"
        clickHandler={[Function]}
        divClass={null}
      />
    </div>
    <hr />
    <div
      className="modal-body"
    >
      <MenuModal />
    </div>
  </div>
</div>
`;

exports[`Modal should render correctly when type prop is orderCanceledMsg 1`] = `
<div
  className="modal show"
  id="meal-modal"
>
  <div
    className="modal-content"
  >
    <div
      className="modal-header"
    >
      <div
        className="modal-title"
      >
        <h3
          id="modal-title-h3"
        />
      </div>
      <CloseIcon
        btnID="modal-close-icon"
        clickHandler={[Function]}
        divClass={null}
      />
    </div>
    <hr />
    <div
      className="modal-body"
    >
      <p
        className="text-center"
      >
        Order Canceled Successfully
      </p>
    </div>
  </div>
</div>
`;

exports[`Modal should render correctly when type prop is orderSuccessMsg 1`] = `
<div
  className="modal show"
  id="meal-modal"
>
  <div
    className="modal-content"
  >
    <div
      className="modal-header"
    >
      <div
        className="modal-title"
      >
        <h3
          id="modal-title-h3"
        />
      </div>
      <CloseIcon
        btnID="modal-close-icon"
        clickHandler={[Function]}
        divClass={null}
      />
    </div>
    <hr />
    <div
      className="modal-body"
    >
      <p
        className="text-center"
      >
        Thank you for your order. Your belly will be filled up shortly
      </p>
    </div>
  </div>
</div>
`;

exports[`Modal should render null when type prop is unknown 1`] = `
<div
  className="modal show"
  id="meal-modal"
>
  <div
    className="modal-content"
  >
    <div
      className="modal-header"
    >
      <div
        className="modal-title"
      >
        <h3
          id="modal-title-h3"
        />
      </div>
      <CloseIcon
        btnID="modal-close-icon"
        clickHandler={[Function]}
        divClass={null}
      />
    </div>
    <hr />
    <div
      className="modal-body"
    />
  </div>
</div>
`;
